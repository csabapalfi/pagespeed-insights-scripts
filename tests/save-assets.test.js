const {resolve} = require('path');
const {writeFileSync} = require('fs');
const {saveAssets: lhSaveAssets} = require('lighthouse/lighthouse-core/lib/asset-saver');
const {saveAssets} = require('../lib/save-assets');

jest.mock('path');
jest.mock('fs');
jest.mock('lighthouse/lighthouse-core/lib/asset-saver');


describe('save-assets', () => {
  const index = 1;
  const audits = {};
  const result = {audits};
  const resolvedPath = 'resolvedPath';

  afterEach(() => { 
    jest.resetAllMocks();
  });

  function expectToSaveReport() {
    expect(resolve).toHaveBeenCalledTimes(1);
    expect(resolve.mock.calls[0])
      .toEqual([process.cwd(), index]);

    expect(writeFileSync).toHaveBeenCalledTimes(1);
    expect(writeFileSync.mock.calls[0]).toMatchSnapshot();
  }

  it('saves report only if no artifacts', async () => {
    resolve.mockReturnValue(resolvedPath);
    
    await saveAssets(index, result);

    expectToSaveReport();
  });

  it('saves report and artifacts', async () => {
    const artifacts = {};
    resolve.mockReturnValue(resolvedPath);
    
    await saveAssets(1, result, artifacts);

    expectToSaveReport();

    expect(lhSaveAssets).toHaveBeenCalledTimes(1);
    expect(lhSaveAssets.mock.calls[0][0]).toBe(artifacts);
    expect(lhSaveAssets.mock.calls[0][1]).toBe(audits);
    expect(lhSaveAssets.mock.calls[0][2]).toBe(resolvedPath);
  });
});