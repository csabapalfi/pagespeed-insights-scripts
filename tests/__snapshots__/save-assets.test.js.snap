// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`save-assets saves report and artifacts 1`] = `
Array [
  "resolvedPath.report.json",
  "{
  \\"audits\\": {}
}",
  "utf-8",
]
`;

exports[`save-assets saves report only if no artifacts 1`] = `
Array [
  "resolvedPath.report.json",
  "{
  \\"audits\\": {}
}",
  "utf-8",
]
`;
